interface/src/app/node/node.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * "adage.gene.network" module.
+ */
+
+angular.module('adage.node', [
+  'ui.router',
+  'placeholders',
+  'ui.bootstrap'
+])
+
+.config(function config($stateProvider) {
+  $stateProvider.state('node', {
+    url: '/node/{id:int}',
+    views: {
+      main: {
+        templateUrl: 'node/node.tpl.html',
+        controller: 'NodeCtrl as ctrl'
+      }
+    },
+    data: {pageTitle: 'Node Information'}
+  });
+})
+
+.factory('NodeInfo', ['$resource', '$stateParams',
+  function($resource) {
+    return $resource('/api/v0/node/:id');
+  }
+])
+
+.factory('HeavyGenes', ['$resource', function($resource) {
+  return $resource('/api/v0/participation/');
+}])
+
+.controller('NodeCtrl', ['NodeInfo', '$stateParams', '$log',
+  function NodeController(NodeInfo, $stateParams, $log) {
+    var self = this;
+    if (!$stateParams.id) {
+      self.statusMessage = 'Please specify node ID in the URL.';
+      return;
+    }
+    self.id = $stateParams.id;
+    self.statusMessage = 'Connecting to the server ...';
+    NodeInfo.get(
+      {id: self.id},
+      function success(response) {
+        self.name = response.name;
+        self.mlmodel = response.mlmodel.title;
+        self.statusMessage = '';
+      },
+      function error(err) {
+        $log.error('Failed to get node information: ');
+        self.statusMessage = 'Failed to get node information from server';
+      }
+    );
+  }
+])
+
+.directive('highWeightGenes', ['HeavyGenes', '$log',
+  function(HeavyGenes, $log) {
+    return {
+      templateUrl: 'node/heavy_genes.tpl.html',
+      scope: {
+        nodeId: '='
+      },
+      link: function($scope) {
+        $scope.queryStatus = 'Connecting to the server ...';
+        HeavyGenes.get(
+          {node: $scope.nodeId, limit: 0},
+          function success(response) {
+            $scope.genes = [];
+            var i = 0, n = response.objects.length;
```

Comment:
This is actually my personal preference to fight with the infamous **variable hoisting** in JavaScript. I am sure you know that a line like this:
```
var foo = 123;
for (var i = 0; i < 10; ++i) { 
  // ... 
}
```
is equivalent to:
```
var foo = 123;
var i;
for (i = 0; i < 10; ++i) { 
  // ... 
}
```
So I put the variable declarations and definitions at the beginning to try to emphasize all declared variables in this function.